snaxc/tools/snax_opt_main.py

```diff
@@ -38,8 +20,9 @@ def __init__(
         self.available_passes = {}
         self.available_targets = {}
 
-        self.ctx = Context()
+        self.ctx: AccContext = AccContext()
         self.register_all_dialects()
+        self.register_all_accelerators()
```

Comment:
This would be the main contribution of this PR, the fact that here we can now choose which accelerators are actually registered :)